src/main/scala/no/gdl/bookapi/service/search/IndexService.scala

```diff
@@ -0,0 +1,296 @@
+/*
+ * Part of GDL book_api.
+ * Copyright (C) 2017 Global Digital Library
+ *
+ * See LICENSE
+ */
+
+package no.gdl.bookapi.service.search
+
+import java.text.SimpleDateFormat
+import java.util.{Calendar, UUID}
+
+import com.sksamuel.elastic4s.alias.{AddAliasActionDefinition, AliasActionDefinition, RemoveAliasActionDefinition}
+import com.sksamuel.elastic4s.analyzers._
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.indexes.IndexDefinition
+import com.sksamuel.elastic4s.mappings.{MappingDefinition, TextFieldDefinition}
+import com.sksamuel.elastic4s.{IndexAndType, RefreshPolicy}
+import com.typesafe.scalalogging.LazyLogging
+import io.digitallibrary.language.model.LanguageTag
+import no.gdl.bookapi.BookApiProperties
+import no.gdl.bookapi.integration.ElasticClient
+import no.gdl.bookapi.model.Language._
+import no.gdl.bookapi.model.api.{GdlSearchException, LocalDateSerializer}
+import no.gdl.bookapi.model.domain
+import no.gdl.bookapi.model.domain.Translation
+import no.gdl.bookapi.repository.{BookRepository, TranslationRepository}
+import no.gdl.bookapi.service.ConverterService
+import org.elasticsearch.ElasticsearchException
+import org.elasticsearch.index.IndexNotFoundException
+import org.json4s.DefaultFormats
+import org.json4s.native.Serialization.write
+
+import scala.util.{Failure, Success, Try}
+
+trait IndexService extends LazyLogging {
+  this: ElasticClient with ConverterService with TranslationRepository with BookRepository =>
+  val indexService: IndexService
+
+  class IndexService {
+    implicit val formats = DefaultFormats + LocalDateSerializer
+
+    def indexDocument(translation: Translation): Try[Translation] = {
+      val availableLanguages: Seq[LanguageTag] = translationRepository.languagesFor(translation.bookId)
+      val book: Option[domain.Book] = bookRepository.withId(translation.bookId)
+      val source = write(converterService.toApiBook(Some(translation), availableLanguages, book))
+
+      esClient.execute(
+        indexInto(BookApiProperties.searchIndex(translation.language), BookApiProperties.SearchDocument)
+          .id(translation.id.toString)
+          .source(source)
+      ) match {
+        case Success(_) => Success(translation)
+        case Failure(failure) => errorHandler(Failure(failure))
+      }
+    }
+
+    def indexDocuments(translationList: List[Translation], indexName: String): Try[Int] = {
+      var actions = List[IndexDefinition]()
```

Comment:
This can be done with a for comprehension yielding `IndexDefinition`s instead of mutating `actions`